Add support for querying status of card and epic (parent)

jangelsb opened the pull request:
* Add support for querying status of card and epic (parent)

Subtitle now looks like this:

`XYZ-1234 (In Development) - 2025 Maintenance`

jangelsb committed:
* Add support for querying status of card and epic (parent)
* Add dictionary safety

caleb531 commented on jfa/list_issues.py:
```diff
@@ -56,20 +56,32 @@ def get_issue_url(issue_key: str) -> str:
-        "subtitle": f"{issue['key']} ({status_name})",
-        "arg": issue["id"],
+        "title": issue_summary,
+        "subtitle": subtitle + " (view in Jira)",
```
@jangelsb I removed the "(view in Jira)" suffix recently in 2ed6c418ea5ecebf1f5bf210e2f7f348bfc27851, so please make sure your fork is operating on the latest tip of the `main` branch.

jangelsb replied:
Hey @caleb531 - sorry for the latest response and update. I just got my branch rebased on yours and all up to date! 

I wasn't aware of how `TypedDict` worked - so I *think* I updated all that correctly - curious to see if we get a green build 🤞

caleb531 reviewed (changes requested):
Can you please explain more of how this querying of the parent card is useful? What is a typical use case here?

Also, my CI is reporting the following type errors:

```
jfa/list_issues.py:65: error: "object" has no attribute "get"  [attr-defined]
jfa/list_issues.py:66: error: "object" has no attribute "get"  [attr-defined]
jfa/list_issues.py:77: error: Missing key "issue_status" for TypedDict "ResultVariables"  [typeddict-item]
jfa/list_issues.py:77: error: Extra keys ("parent_key", "parent_summary") for TypedDict "ResultVariables"  [typeddict-unknown-key]
```

jangelsb committed:
* Update to use latest code
* Keep the same subtitle format but add ` - {Epic}`

jangelsb committed:
* Attempt to fix cli issues but updating the types
* Revert back the dictionary safety since the types are explicit

jangelsb commented:
> Can you please explain more of how this querying of the parent card is useful? What is a typical use case here?
> 
> Also, my CI is reporting the following type errors:
> 
> ```
> jfa/list_issues.py:65: error: "object" has no attribute "get"  [attr-defined]
> jfa/list_issues.py:66: error: "object" has no attribute "get"  [attr-defined]
> jfa/list_issues.py:77: error: Missing key "issue_status" for TypedDict "ResultVariables"  [typeddict-item]
> jfa/list_issues.py:77: error: Extra keys ("parent_key", "parent_summary") for TypedDict "ResultVariables"  [typeddict-unknown-key]
> ```

Hey @caleb531 

I wasn't aware of how `TypedDict` worked - so I *think* I updated all that (hopefully) correctly - curious to see if we get a green build 🤞

I reverted all the dictionary safety since it shouldn't be needed since we are explicitly setting the types

For your first question - I updated my local version of this workflow to show the parent (aka an Epic) and I thought it was really helpful. I am a tech lead for a software company and we deal a lot with Epics. Every card needs to have an epic, etc. And when I am moving cards around or creating cards, or just looking at old cards, it is nice to know which Epic it is at a glance

If it adds too much bloat - we don't have to merge this in - but wanted to share just incase!

jangelsb committed:
* Fix lint error

jangelsb commented:
Woo! Green build and I tested it and it works

Subtitle looks like this

`XYZ-1234 (In Development) - 2025 Maintenance`

jangelsb commented:
> @jangelsb This looks great! Does this work for subtasks too? As in, does the parent _have_ to be an epic? Or could it be a regular task?

@caleb531 I am not 100% since we don't use subtasks 😅 but this is the official `parent` property in the jira API - so it seems totally reusable it could work for subtasks (which would be super cool!). I just did a little research and looks like jira changed the property key from `epic` -> `parent` in the past. And ChatGPT (take with a grain of salt) says that `parent` can be used with subtasks 

> Also, was just curious if you observed any performance differences with this.

I havet noticed any - based on my own testing, it seems like they store all this information together on their side to keep it fast

But I'll set a reminder for myself on Monday and compare your old build to my modified build when searching and see if I can notice

caleb531 commented on jfa/types.py:
```diff
@@ -3,6 +3,15 @@
 from typing import Optional, TypedDict
 
 
+class ParentFields(TypedDict):
```
@jangelsb I know this is nitpicky, but for clarity, my preference would be `ParentIssue*`, not `Parent*`, if you could please make that tweak.

```suggestion
class ParentIssueFields(TypedDict):
```

caleb531 replied:
Okay, updated in 35a0843b39d7ec9c0c97d63471ed09efbd81f662.

caleb531 committed:
Rename Parent* types to ParentIssue*

caleb531 commented on jfa/types.py:
```diff
@@ -17,6 +26,7 @@ class IssueFields(TypedDict):
     summary: str
     issuetype: IssueType
     status: IssueStatus
+    parent: Optional[Parent]
```
Okay, decided to tweak this myself in 35a0843b39d7ec9c0c97d63471ed09efbd81f662.

caleb531 commented on jfa/types.py:
```diff
@@ -3,6 +3,15 @@
+    summary: str
+
+
+class Parent(TypedDict):
```
Okay, updated in 35a0843b39d7ec9c0c97d63471ed09efbd81f662.

caleb531 commented:
@jangelsb Okay, I decided to tweak the class names myself, so everything looks good to me now! I've gone ahead and merged this, and will publish a release sometime in the near future.

jangelsb commented:
Hey @caleb531 sorry I didn't see these requests. Thanks for updating it, I appreciate it!
